### PR TITLE
chore: refresh OTel dependencies

### PR DESCRIPTION
**Pull request by seemk**
Latest OTel only supports Ruby 3.x, think it's a good time to evaluate dropping support for 2.x.

### CODE REVIEW

**Comment by seemk** [comment]
Figuring out the bundler dep issue :thinking:

**Comment by seemk** [comment]
The new OTel deps require Ruby 3.0, do we need to support Ruby <3.0?

File: examples/basic/console.rb [hunk 1]
```diff
@@ -7,12 +7,10 @@
   source "https://rubygems.org"
 
   gem "splunk-otel", path: "../../"
-  gem "opentelemetry-exporter-otlp", "~> 0.21.0"
 end
 
 require "splunk/otel"
 require "splunk/otel/logging"
-require "opentelemetry/exporter/otlp"
```
Comment (tsloughter-splunk) [inline examples/basic/console.rb:15]: Why is this removed?

**Comment by tsloughter-splunk** [comment]
If the latest ruby otel requires ruby 3+ we may have to also.

File: examples/basic/console.rb (hunk 1 above)
Comment (seemk) [reply examples/basic/console.rb:15]: Don't think the example uses it in any way, `opentelemetry-exporter-otlp` is brought in via the distro anyway

Comment (tsloughter-splunk) [reply examples/basic/console.rb:15]: Ah right, ok.

**Review by johnbley — APPROVED** [review]
